raster/rt_pg/rtpostgis.c

```diff
@@ -239,12 +241,215 @@ rt_pg_debug(const char *fmt, va_list ap)
     ereport(DEBUG1, (errmsg_internal("%s", msg)));
 }
 
+static char *
+rt_pg_options(const char* varname)
+{
+	char optname[128];
+	char *optvalue;
+	snprintf(optname, 128, "postgis.%s", varname);
+	/* GetConfigOptionByName(name, found_name, missing_ok) */
+	optvalue = GetConfigOptionByName(optname, NULL, true);
+	if (optvalue && strlen(optvalue) == 0)
+		return NULL;
+	else
+		return optvalue;
+}
+
+/* ---------------------------------------------------------------- */
+/*  GDAL allowed config options for VSI filesystems */
+/* ---------------------------------------------------------------- */
+
+stringlist_t *vsi_option_stringlist = NULL;
+
+
+#if POSTGIS_GDAL_VERSION < 23
+
+/*
+* For older versions of GDAL we  have extracted the list of options
+* that were available at the 2.2 release and use that
+* as our set of allowed VSI network file options.
+*/
+static void
+rt_pg_vsi_load_all_options(void)
+{
+	const char *gdaloption;
+	const char *gdaloptions[] = {
```

Comment:
For the array to be fully read-only:
```suggestion
	const char * const gdaloptions[] = {
```